developer_docs/how_to_create_a_plugin.asciidoc

```diff
@@ -1678,6 +1678,42 @@ The settings are strongly typed and you have to define it.
 The basic types supported by Foreman are: `:boolean`, `:integer`, `:float`, `:string`, `:text`, `:hash`, `:array`.
 The `:text` type supports markdown and usage of such setting should expect markdown syntax when using it.
 
+==== Validate setting value
+
+To validate setting value, you can use API, that tries to mimic the API of ActiveRecord.
+We can use most of the perks offered by ActiveRecord, only defining on setting name instead of attribute.
+We are adding just some shothands like direct regexp validations.
+The attribute is always `value`, you can't validate anything else as it is the only user input.
+Custom validation methods allow you to run validations against other attributes, but please dont! :)
+
+You have two ways of define the validations:
+* inline with setting definition
+* using `validates` of `validates_with` methods
+
+There is special option `allow_blank`, that needs to be defined on setting definition.
+This also allow blank default value.
+
+[source, ruby]
+....
+settings do
+  category(:cfgmgmt) do
+    # Following definitions are missing defaults and full names for simplification
+
+    setting(:blank_setting, type: :string, description: 'Unnecessary setting', allow_blank: true)
+
+    setting(:cool_setting, type: :string, description: 'Setting with only cool values', validate: /^cool.*/, allow_blank: true)
+
+    setting(:cooler_puppet, type: :integer, description: 'Use Puppet that goes to 11', validate: ->(s) { s.value >= 11 })
```

Comment:
Shouldn't this be up to and including 11, but not higher?
```suggestion
    setting(:cooler_puppet, type: :integer, description: 'Use Puppet that goes to 11', validate: ->(s) { s.value <= 11 })
```

Reply:
That depends, I was trying to suggest you can be even cooler then 11 :thinking:

Reply:
The original reference of "goes to 11" is an amplifier that has a volume that goes to 11. So it's louder than all the other that go to 10. Of course it's all a joke, but there's a serious note to this: if you understand the joke, you can reason that `validate` will raise an error if the condition evaluates to true instead of false.